### PR TITLE
Issue #7564: Added example for default config

### LINKED ISSUE #7564: Add default config example for ConstantName
split from #5832

update AbstractClassName to make examples for all configs shown in documentation
https://checkstyle.org/config_naming.html#ConstantName
this check need default config and example of code that is violates and expect as correct.

See original issues for examples on how to make update.

------------

PLEASE READ https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-examples

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Resolves #7564 Add Default config example for ConstantName.

### CODE REVIEW

**Comment by Vyom-Yadav** [comment]
Sorry, I had to close the earlier pull request, there was some problem.

File: src/xdocs/config_naming.xml [hunk 1]
```diff
@@ -911,8 +929,8 @@ class MyClass {
         </p>
         <source>
 &lt;module name=&quot;ConstantName&quot;&gt;
-  &lt;property name="applyToPublic" value="false"/&gt;
-  &lt;property name="applyToProtected" value="false"/&gt;
+&lt;property name="applyToPublic" value="false"/&gt;
+&lt;property name="applyToProtected" value="false"/&gt;
```
Comment (strkkk) [inline src/xdocs/config_naming.xml:933]: revert this, you should not change this indentation

File: src/xdocs/config_naming.xml [hunk 2]
```diff
@@ -903,6 +903,24 @@ class MyClass {
...
+}
+        </source>
+        <p>
+         The following configuration is applicable by default.
+        </p>
+        <source>
+&lt;module name=&quot;ConstantName&quot;&gt;
+&lt;/module&gt;
+        </source>
+        <p>Code Example:</p>
```
Comment (strkkk) [inline src/xdocs/config_naming.xml:915]: `Code Example:` -> Example:

File: src/xdocs/config_naming.xml [hunk 3]
```diff
@@ -903,6 +903,24 @@ class MyClass {
...
   final static int myselfConstant = 1; // violation, name 'myselfConstant' must match pattern
                                        // '^log(ger)?$|^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+}
+        </source>
+        <p>
+         The following configuration is applicable by default.
+        </p>
+        <source>
+&lt;module name=&quot;ConstantName&quot;&gt;
+&lt;/module&gt;
```
Comment (strkkk) [inline src/xdocs/config_naming.xml:913]: you do not need config example here. There is already config example at line 881, you only need to add code example to it.

**Review by strkkk — CHANGES_REQUESTED** [review]

File: src/xdocs/config_naming.xml (hunk 1 above)
Comment (Vyom-Yadav) [reply src/xdocs/config_naming.xml:933]: It was showing an error that's why I changed.

File: src/xdocs/config_naming.xml (hunk 2 above)
Comment (Vyom-Yadav) [reply src/xdocs/config_naming.xml:915]: ok, I will do it.

File: src/xdocs/config_naming.xml (hunk 3 above)
Comment (Vyom-Yadav) [reply src/xdocs/config_naming.xml:913]: So I shall delete this and add the example below line 881?

File: src/xdocs/config_naming.xml (hunk 2 above)
Comment (Vyom-Yadav) [reply src/xdocs/config_naming.xml:915]: Also, shall I make another commit? or go to files changed and just edit it over there?

File: src/xdocs/config_naming.xml (hunk 3 above)
Comment (Vyom-Yadav) [reply src/xdocs/config_naming.xml:913]: So what shall I delete or add over here? You want me to delete <source> and its content over here?

**Comment by Vyom-Yadav** [comment]
It started showing 'This commit does not belong to any branch on this repository' for my first commit when I added another commit, I was going to rebase them and squash them, what shall I do?

**Comment by strkkk** [comment]
@Vyom-Yadav few notes
1. Please reply to review comments only if you have questions or you corrected item (no "ok", "will do" or something like this)
2. Do not push many commits, always keep one commit (amend existing or squash several into one)

File: src/xdocs/config_naming.xml (hunk 3 above)
Comment (strkkk) [reply src/xdocs/config_naming.xml:913]: remove config example and code example to existing default config example

File: src/xdocs/config_naming.xml (hunk 2 above)
Comment (strkkk) [reply src/xdocs/config_naming.xml]: replied below

File: src/xdocs/config_naming.xml (hunk 1 above)
Comment (strkkk) [reply src/xdocs/config_naming.xml]: there should be no error, please revert and share error message

**Comment by Vyom-Yadav** [comment]
@strkkk The tests that are failing are only due to commit name, if I change commit name, will these tests re-run? Is there actually a need to change commit name, just one space is missing?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java [hunk 4]
```diff
@@ -61,10 +61,23 @@
  * <p>
  * To configure the check:
  * </p>
+ * <p>
+ * The following configuration is applicable by default.
+ * </p>
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java:66]: Please remove this section, there is already a section above

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java [hunk 5]
```diff
@@ -61,10 +61,23 @@
...
  * </pre>
- *
+ * <p>Example:</p>
+ * <pre>
+ * class MyClass {
+ *   public final static int FIRST_CONSTANT1 = 10; // OK
+ *   protected final static int SECOND_CONSTANT2 = 100; // OK
+ *   final static int third_Constant3 = 1000; // violation, name 'third_Constant3' must
+ *                                           // match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+ *   private final static int fourth_Constant4 = 50;// violation, name 'fourth_Constant4' must match
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java:77]: `private final static int fourth_Constant4 = 50;//`
Please add space here ------------------------------^

**Review by strkkk — CHANGES_REQUESTED** [review]
Much better now, few minors

**Comment by Vyom-Yadav** [comment]
@strkkk have done all the changes you asked for.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java (hunk 5 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java:77]: I added but it showed that limit of characters exceeded(100), mine was 101 so I used a backspace

Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java:77]: I had added them in the beginning and it showed an error that line length exceeded(100 is max) mine was 101, so I used a backspace. I rechecked it but the error was still there.
[ERROR] [checkstyle] [ERROR] E:\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\naming\ConstantNameCheck.java:74: Line is longer than 100 characters (found 101). [LineLength]

Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java:77]: Please change constant name to other name, which is shorter, to satisfy line length check

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java (hunk 4 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java:66]: @Vyom-Yadav ^^^

**Comment by Vyom-Yadav** [comment]
@strkkk I have done all the changes, kindly review.

**Comment by Vyom-Yadav** [comment]
@strkkk Why this error is coming? I am unable to find the reason.

**Comment by Vyom-Yadav** [comment]
github generate site

**Comment by romani** [comment]
@Vyom-Yadav , please keep replying "done" to each addressed review item.

Please rebase your branch on most latest our master to resolve Travis problem. It is fixed already.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java (hunk 5 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java (hunk 4 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java]: done.

**Comment by Vyom-Yadav** [comment]
Github, rebase

**Comment by Vyom-Yadav** [comment]
How to fix it, I had just updated my master branch with upstream

**Comment by Abhishek-kumar09** [comment]
Github, rebase

**Comment by Vyom-Yadav** [comment]
Github, rebase

**Comment by Vyom-Yadav** [comment]
@romani
`Commit messages from github: "Issue #7564: Added example for default config"...`
`Issue number: #7564`
`PR Description grepped:`
`Please put a reference to an Issue in the PR description,`
`this will bind the Issue to your PR in Github`
`PR validation failed.`
it asks for PR description but it is already there.

**Comment by Abhishek-kumar09** [comment]
image
This is your PR description. It is currently empty. Please put the Pr description.
like: Resolves #7564

**Comment by Vyom-Yadav** [comment]
> image
> This is your PR description. It is currently empty. Please put the Pr description.
> like: Resolves #7564

I did it, now how to re run the tests?

**Comment by romani** [comment]
@strkkk , please finish review

**Review by strkkk — APPROVED** [review]

**Comment by romani** [comment]
Github, generate web site

**Pull request merged**